src/util/worker/mod.rs

```diff
@@ -80,21 +82,53 @@ impl<T: Display, R: Runnable<T>> BatchRunnable<T> for R {
     }
 }
 
+#[derive(Clone)]
+struct ScheduleContext {
+    counter: Arc<AtomicUsize>,
+    #[cfg(test)]
+    pause: Arc<(Mutex<bool>, Condvar)>,
+}
+
+#[cfg(not(test))]
+impl ScheduleContext {
+    fn new() -> ScheduleContext {
+        ScheduleContext { counter: Arc::new(AtomicUsize::new(0)) }
+    }
+
+    fn pause_if_neccessary(&self) {}
+}
+
+#[cfg(test)]
+#[allow(mutex_atomic)]
+impl ScheduleContext {
+    fn new() -> ScheduleContext {
+        ScheduleContext {
+            counter: Arc::new(AtomicUsize::new(0)),
+            pause: Arc::new((Mutex::new(false), Condvar::new())),
+        }
+    }
+
+    fn pause_if_neccessary(&self) {
```

Comment:
I think `try_pause` should return a result or bool to indicate if it's paused. Seem not suitable here.